pandas/tests/generic/test_generic.py

```diff
@@ -897,3 +897,13 @@ def test_flags_identity(self, as_frame):
         assert s.flags is s.flags
         s2 = s.copy()
         assert s2.flags is not s.flags
+
+    def test_slice_incomplete_datetimes(self):
+        # GH36953 and GH35509
+        increment = pd.Series(
+            pd.to_timedelta([30.0, 30.1, 29.9, 30.0], unit="s")
+        ).cumsum()
+        increment[-1] = None
+        index = pd.Timestamp("2012-01-01 09:00") + increment
+        df = pd.Series(range(len(index)), index=index).to_frame()
+        assert len(df["2012-01-01":"2012-01-05"]) == 1000
```

Comment:
can you construct the expected result and compare against that with `tm.assert_frame_equal`?

Reply:
done